pyi.py

```diff
@@ -2110,8 +2135,14 @@ def _visit_function(self, node: ast.FunctionDef | ast.AsyncFunctionDef) -> None:
             ):
                 self.error(statement, Y010)
 
+        decorator_names = {
+            decorator.id
+            for decorator in node.decorator_list
+            if isinstance(decorator, ast.Name)
+        }
```

Comment:
I'd personally use a helper function (e.g. `decorator_names(node)`) instead of passing the names down to the functions involved. I don't think this should have a considerable performance impact, considering the relative rarity of decorators.